app/src/main/java/com/android/unio/ui/settings/Settings.kt

```diff
@@ -90,6 +141,29 @@ fun SettingsContainer() {
                     context.getString(R.string.settings_notifications_content_description))
           },
           defaultValue = true)
+      preference(
+          modifier = Modifier.testTag(AppPreferences.LOCATION_PERMISSION),
+          key = AppPreferences.LOCATION_PERMISSION,
+          title = { Text(context.getString(R.string.settings_location_title)) },
+          summary = {
+            if (locationPermissions.allPermissionsGranted) {
+              Text(context.getString(R.string.settings_location_summary_granted))
+            } else {
+              Text(context.getString(R.string.settings_location_summary_denied))
+            }
+          },
+          icon = {
+            Icon(
+                imageVector = Icons.Default.LocationOn, contentDescription = "Location Permissions")
```

Comment:
You might want to move the string of contentDescription to the file strings.xml, I think under the accessbility and content description